### DISCUSSION TITLE
About Replication

### DISCUSSION THREAD

**Discussion by c1kk** [ravendb/ravendb#13384] [Q&A]
I have a question about replication in distributed environment. When i create database have a set up replication for node.  I don't know it is a replication in theory. As i know about replication is A create database and B have that database too and if A have a problem with database, then B still keep that database use it normally.

**Comment by ayende** [comment]
Yes, that is correct. We discuss that briefly here: https://ravendb.net/why-ravendb/high-availability

For more details, see: https://ravendb.net/learn/inside-ravendb-book/reader/4.0/6-ravendb-clusters#replication-of-data-in-a-database-group

**Reply by c1kk** [reply]
What about External Replication Task ? How is it different from replication default's system when i create DB.

**Reply by ayende** [reply]
For replicas inside the same database (internal replication), RavenDB handles all data replication, failover, etc.
The cluster handles things like indexes, tasks, etc. 

You can also define external replication (between different databases or different clusters). There is no automatic failover between those and just the data is replicated, nothing else.

**Reply by c1kk** [reply]
![GetDocImage](https://user-images.githubusercontent.com/92294829/147841963-941d4d4b-0c58-4d1f-ad6a-48ce1fb09df7.png)
About replication in External, a replication to db3 (number 2) is a replication of node C. If data of C have problem (can't use), can data of Replication to db3 used to replace it while A is repairing ? or in case data of A is lost , data of replication is lost too ?

**Reply by ayende** [reply]
This is an external replication task that is currently assigned to node C. If node C is down, the cluster will reassign this task to another node.
If node C falls too far behind, the same will happen.

You don't have to do anything, this happens automatically.
